examples/coap-server.c

```diff
@@ -566,49 +759,83 @@ hnd_put(coap_context_t *ctx UNUSED_PARAM,
     /* Need to do this as coap_get_uri_path() created it */
     coap_delete_string(uri_path);
     response->code = COAP_RESPONSE_CODE(204);
-    dynamic_entry[i].created = 0;
-    coap_resource_notify_observers(dynamic_entry[i].resource, NULL);
   }
 
   resource_entry = &dynamic_entry[i];
 
   if (coap_get_block(request, COAP_OPTION_BLOCK1, &block1)) {
     /* handle BLOCK1 */
-    if (coap_get_data(request, &size, &data) && (size > 0)) {
-      size_t offset = block1.num << (block1.szx + 4);
-      coap_string_t *value = resource_entry->value;
-      if (offset == 0) {
-        if (value) {
-          coap_delete_string(value);
-          value = NULL;
-        }
+    coap_cache_entry_t *cache_entry = coap_cache_get_by_pdu(session,
+                                                            request,
+                                                 COAP_CACHE_IS_SESSION_BASED);
+    size_t offset = block1.num << (block1.szx + 4);
+    coap_binary_t *data_so_far;
+
+    if (!cache_entry && block1.num == 0) {
+      cache_entry = coap_new_cache_entry(session, request, 0, 1, -1);
```

Comment:
Here, the symbolic names for the enums should be used as well.

Reply:
Fixed, and again -1 is now 0 as unsigned int.